### ISSUE TITLE
the result with best_model and train with get_best_hyperparameters is different

### ISSUE THREAD

**Issue by melitong** [keras-team/keras-tuner#548]
Hello, i've been used this keras-tuner for a while, but when i train my data with tuner.get_best_hyperparameters(num_trials=1)[0], the result (accuracy) is different with the result from tuner.get_best_models(num_models=1)[0]. 

model summary from tuner.get_best_hyperparameters(num_trials=1)[0] and tuner.get_best_models(num_models=1)[0] is the same? 

could anyone give me advice to solve this?

Thank You

**Comment by douira** [comment]
`get_best_models` returns the best model in it's best epoch while `get_best_hyperparameters` returns the set of hyperparamters that resulted in the best model.

**Comment by melitong** [comment]
oke, thank you for the explanation
